cli/cmd/uninstall/uninstall.go

```diff
@@ -0,0 +1,419 @@
+	// use a consistent method to target the right cluster for both Helm SDK and non Helm SDK calls.
+	if c.kubernetes == nil {
+		restConfig, err := settings.RESTClientGetter().ToRESTConfig()
+		if err != nil {
+			c.UI.Output("retrieving Kubernetes auth: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+		c.kubernetes, err = kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			c.UI.Output("initializing Kubernetes client: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	// Setup logger to stream Helm library logs
+	var uiLogger = func(s string, args ...interface{}) {
+		logMsg := fmt.Sprintf(s, args...)
+		c.UI.Output(logMsg, terminal.WithLibraryStyle())
+	}
+
+	c.UI.Output("Existing Installation", terminal.WithHeaderStyle())
+
+	// Search for Consul installation by calling `helm list`. Depends on what's already specified.
+	actionConfig := new(action.Configuration)
+	actionConfig, err := c.initActionConfig(actionConfig, c.flagNamespace, settings, uiLogger)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	found, foundReleaseName, foundReleaseNamespace, err := c.findExistingInstallation(actionConfig)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+	if !found {
+		c.UI.Output("No existing Consul installations.", terminal.WithSuccessStyle())
+	} else {
+		c.UI.Output("Existing Consul installation found.", terminal.WithSuccessStyle())
+		c.UI.Output("Consul Un-Install Summary", terminal.WithHeaderStyle())
+		c.UI.Output("Installation name: %s", foundReleaseName, terminal.WithInfoStyle())
+		c.UI.Output("Namespace: %s", foundReleaseNamespace, terminal.WithInfoStyle())
+
+		// Prompt for approval to uninstall Helm release.
+		if !c.flagAutoApprove {
+			confirmation, err := c.UI.Input(&terminal.Input{
+				Prompt: "Proceed with uninstall? (y/N)",
+				Style:  terminal.InfoStyle,
+				Secret: false,
+			})
+			if err != nil {
+				c.UI.Output(err.Error(), terminal.WithErrorStyle())
+				return 1
+			}
+			confirmation = strings.TrimSuffix(confirmation, "\n")
+			if !(strings.ToLower(confirmation) == "y" || strings.ToLower(confirmation) == "yes") {
+				c.UI.Output("Un-install aborted. To learn how to customize the uninstall, run:\nconsul-k8s uninstall --help", terminal.WithInfoStyle())
+				return 1
+			}
+		}
+		// Actually call out to `helm delete`.
+		actionConfig, err = c.initActionConfig(actionConfig, foundReleaseNamespace, settings, uiLogger)
+		if err != nil {
+			c.UI.Output(err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+		uninstaller := action.NewUninstall(actionConfig)
+		res, err := uninstaller.Run(foundReleaseName)
+		if err != nil {
+			c.UI.Output("unable to uninstall: %s", err, terminal.WithErrorStyle())
+			return 1
+		} else if res != nil && res.Info != "" {
+			c.UI.Output("uninstall result: %s", res.Info, terminal.WithErrorStyle())
```

Comment:
Thanks for catching the result being printed with error, that's not an error case you're right. I updated that. 

I have a bunch of helper functions throughout that return errors, and then if that function returns an error we print it out to the UI with error style. Since those errors will be lowercased, I was thinking it'd make more sense if any errors we output were lowercased. WDYT?

Reply:
Oh I see, that makes sense! I think I have a slight bias toward consistency but that's fine too. I'm thinking about how to make it more obvious in the code in case someone else comes in and thinks the same thing. I think a comment before every error output is not feasible, but maybe we can create a helper function for error outputs? Doesn't have to be part of this PR though!

Reply:
yeah great point, added an asana for this. I plan to address this while doing upgrade command https://app.asana.com/0/1200500259727746/1201031492148295/f